rust-runtime/aws-smithy-types/src/date_time/mod.rs

```diff
@@ -52,6 +52,9 @@ const NANOS_PER_SECOND_U32: u32 = 1_000_000_000;
 #[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
 pub struct DateTime {
     seconds: i64,
+    /// Subsecond nanos always advances the wallclock time, even for times where seconds is negative
+    ///
+    /// Bigger subsecond nanos => later time
```

Comment:
Consider adding one more example to show that
```
DateTime {
  seconds: -1,
  subsecond_nanos: 1_000_000,
}
```
specifies the same wallclock time as
```
DateTime {
  seconds: 0,
  subsecond_nanos: 0,
}
```